File: lua/triangle/triangle.lua
local triangle = {}

local function sum_of_sides_are_unacceptable(a, b, c)
  return a + b < c or b + c < a or c + a < b
end

local function side_lengths_are_unacceptable(a, b, c)
  return a <= 0 or b <= 0 or c <= 0
end
function triangle.kind(a, b, c)
  if sum_of_sides_are_unacceptable(a, b, c) then error'Input Error' end

  if side_lengths_are_unacceptable(a, b, c) then error'Input Error' end

  if a == b and b == c then
    return "equilateral"
  elseif a == b or b == c or a == c then
    return "isosceles"
  else
    return "scalene"
  end
end

return triangle

File: clojure/triangle/src/triangle.clj
(ns triangle)

(defn is-valid? [s s2 s3]
  (and (>= (+ s s2) s3) (>= (+ s s3) s2) (>= (+ s2 s3) s)
       (every? pos? (list s s2 s3))))

(defn equilateral? [s s2 s3]
  (if (is-valid? s s2 s3)
    (and (= s s2) (= s s3) (= s2 s3))
    false
    ))

(defn isosceles? [s s2 s3]
  (if (is-valid? s s2 s3)
    (or (= s s2) (= s s3) (= s2 s3))
    false))

(defn scalene? [s s2 s3]
  (if (is-valid? s s2 s3)
    (= 3 (count (set (list s s2 s3))))
    false
    ))
